packages/web-components/src/slider/slider.ts

```diff
@@ -267,20 +536,27 @@ export class Slider extends FormAssociatedSlider implements SliderConfiguration
   @attr
   public mode: SliderMode = SliderMode.singleValue;
 
+  constructor() {
+    super();
+
+    this.elementInternals.role = 'slider';
+    this.elementInternals.ariaOrientation = this.orientation ?? SliderOrientation.horizontal;
+  }
+
   /**
    * @internal
    */
   public connectedCallback(): void {
     super.connectedCallback();
 
-    this.proxy.setAttribute('type', 'range');
-
     this.direction = getDirection(this);
+    this.disabled = this.hasAttribute('disabled');
```

Comment:
Can you elaborate why this is needed?

Reply:
If I remove this and add a default value to `disabled` like `@attr(...) public disabled = false;`, when running Playwright, it throws `DOMException: Failed to construct 'CustomElement': The result must not have attributes` for [this test](https://github.com/microsoft/fluentui/blob/0ee179c9193a7aafdb8155d8b68b4b8aeec4a89e/packages/web-components/src/slider/slider.spec.ts#L527).

The error is referring to making changes to an attribute (`disabled`) within the custom element’s constructor. I see the same error in Storybook, when I try to run `document.createElement('fluent-slider')` in the console. I thought this was an issue in FAST Element.

Reply:
> If I remove this and add a default value to `disabled` like `@attr(...) public disabled = false;`, when running Playwright, it throws `DOMException: Failed to construct 'CustomElement': The result must not have attributes` for [this test](https://github.com/microsoft/fluentui/blob/0ee179c9193a7aafdb8155d8b68b4b8aeec4a89e/packages/web-components/src/slider/slider.spec.ts#L527).

Hmmm...we should explore this - we've got several instances where this exists already in the codebase. Can we sync on it tomorrow?

Reply:
Never mind. I found the issue, it's because I was updating `this.tabIndex` in the `setDisabledSideEffects()` method which is called inside `disabledChanged()`, and that sets the `tabindex` attribute. I updated the code, now it's working fine. Thanks for double checking on this!